src/cli-validator/utils/fixProblems.js

```diff
@@ -0,0 +1,212 @@
+const each = require('lodash/each');
+const fs = require('fs');
+const snakeCase = require('snake-case');
+
+// this function prints all of the output
+module.exports = function print(results, originalFile, errorsOnly) {
+  const types = errorsOnly ? ['errors'] : ['errors', 'warnings'];
+
+  const originalJSON = JSON.parse(originalFile);
+
+  const paramsToDelete = {};
+  const paramsToPrepend = {};
+  const newPathNames = {};
+  const definitionsToDelete = {};
+
+  const markForDeletion = path => {
+    const pathstr = path.slice(0, path.length - 1).join(',');
+    if (pathstr in paramsToDelete) {
+      paramsToDelete[pathstr].push(path[path.length - 1]);
+    } else {
+      paramsToDelete[pathstr] = [path[path.length - 1]];
+    }
+  };
+
+  const markForMoving = (path, parent) => {
+    const pathstr = path.slice(0, path.length - 1).join(',');
+    if (
+      !(
+        pathstr in paramsToDelete &&
+        path[path.length - 1] in paramsToDelete[pathstr]
+      )
+    ) {
+      if (pathstr in paramsToPrepend) {
+        paramsToPrepend[pathstr].push(parent[path[path.length - 1]]);
+      } else {
+        paramsToPrepend[pathstr] = [parent[path[path.length - 1]]];
+      }
+    }
+  };
+
+  const unmarkIfMarkedForMoving = (path, parent) => {
+    const pathstr = path.slice(0, path.length - 1).join(',');
+    const compFunc = param =>
+      JSON.stringify(param) === JSON.stringify(parent[path[path.length - 1]]);
+    if (
+      pathstr in paramsToPrepend &&
+      paramsToPrepend[pathstr].findIndex(compFunc) != -1
+    ) {
+      paramsToPrepend[pathstr].splice(
+        paramsToPrepend[pathstr].findIndex(compFunc),
+        1
+      );
+    }
+  };
+
+  const getObjectFromPath = path => {
+    const pathcopy = path.slice(0);
+    let obj = originalJSON;
+    let parent, pParent, ppParent;
+    while (pathcopy.length) {
+      ppParent = pParent;
+      pParent = parent;
+      parent = obj;
+      obj = obj[pathcopy.shift()];
+    }
+    return { obj, parent, pParent, ppParent };
+  };
+
+  // For each problem type and for every problem
+  types.forEach(type => {
+    each(results[type], problems => {
+      problems.forEach(problem => {
+        const message = problem.message.split(':')[0];
+        let path = problem.path;
+        // path needs to be an array to get the line number
+        if (!Array.isArray(path)) {
+          path = path.split('.');
+        }
+        if (path[path.length - 1].includes('[')) {
+          path.push(path[path.length - 1].split(']')[0].split('[')[1]);
+          path[path.length - 2] = path[path.length - 2].split(['['])[0];
+        }
+        const { obj, parent, ppParent } = getObjectFromPath(path);
+        if (message.includes('Schema must have a non-empty description')) {
+          obj['description'] = path[path.length - 1];
```

Comment:
I'm uncomfortable with this fix.  The spirit of a `description` is to _add_ information, and this appears to just repeat information that is already present.

Reply:
Good point, it doesn't really add any information so I'll leave it out.